### PR TITLE
fix(NetworkName): fix form with insufficient rights

### PR DESCRIPTION
**Pull request by stonebuzz**
If current user have no right on ```internet``` 

image

it cannot add / update IP addresses, 
but from ```NetworkName``` form ```IP Addresses``` label is displayed

image

which leads to confusion, see #15990

| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #15990

### CODE REVIEW

File: src/NetworkName.php [hunk 1]
```diff
@@ -475,25 +475,39 @@ public static function showFormForNetworkPort($networkPortID)
...
+        if ($canedit) {
+            echo "<tr class='tab_bg_1'>\n";
+            echo "<td>" . IPAddress::getTypeName(Session::getPluralNumber());
+            IPAddress::showAddChildButtonForItemForm($name, 'NetworkName__ipaddresses', $canedit);
+            echo "</td>";
+            echo "<td>";
+            IPAddress::showChildsForItemForm($name, 'NetworkName__ipaddresses', $canedit);
+            echo "</td>";
+
+           // MoYo : really need to display it here ?
```
Comment (orthagh) [inline src/NetworkName.php:501]: This comment has 10 years, I guess you can profit to delete this part

Comment (stonebuzz) [reply src/NetworkName.php]: agree with that

File: src/NetworkName.php [hunk 2]
```diff
@@ -475,25 +475,31 @@ public static function showFormForNetworkPort($networkPortID)
...
-        echo "<td colspan='2'>&nbsp;</td>";
-
-        echo "</tr>\n";
+        echo "</tr>";
+
+        if ($name->isNewItem()) {
+            if (!$name->canCreate()) {
+                $canedit = false;
+            }
+            $canedit = $name->canUpdate();
```
Comment (trasher) [inline src/NetworkName.php:484]: Why checking for update rigth when adding? (I'ma always a bit lost in glpi acls, but that sounds strange)

File: src/NetworkName.php [hunk 3]
```diff
@@ -475,25 +475,31 @@ public static function showFormForNetworkPort($networkPortID)
...
-       // echo "</td>\n";
-        echo "<td colspan='2'>&nbsp;</td>";
-
-        echo "</tr>\n";
+        echo "</tr>";
+
+        if ($name->isNewItem()) {
+            if (!$name->canCreate()) {
+                $canedit = false;
+            }
```
Comment (cedric-anne) [inline src/NetworkName.php:483]: `$canedit` value will be overriden by the next line.

File: src/NetworkName.php [hunk 4]
```diff
@@ -475,25 +475,31 @@ public static function showFormForNetworkPort($networkPortID)
...
+            $canedit = $name->canUpdate();
+        } else {
+            if (!$name->can($name->getID(), READ)) {
+                $canedit = false;
+            }
```
Comment (cedric-anne) [inline src/NetworkName.php:488]: `$canedit` value will be overriden by the next line.

**Review by cedric-anne — CHANGES_REQUESTED** [review]

**Comment by stonebuzz** [comment]
I simply took over what was used afterwards, 

```php

    public static function showAddChildButtonForItemForm(
        CommonDBTM $item,
        $field_name,
        $canedit = null,
        $display = true
    ) {

        $items_id = $item->getID();

        if (is_null($canedit)) {
            if ($item->isNewItem()) {
                if (!$item->canCreate()) {
                    return false;
                }
                $canedit = $item->canUpdate();
            } else {
                if (!$item->can($items_id, READ)) {
                    return false;
                }

                $canedit = $item->can($items_id, UPDATE);
            }
        }
........
}
``` 

in order to know in advance whether the user can or can't use it.

**Comment by stonebuzz** [comment]
the last commit simplifies the check

**Review by cedric-anne — APPROVED** [review]

**Review by trasher — APPROVED** [review]

**Pull request merged**